skyllh/core/analysis_utils.py

```diff
@@ -84,6 +85,78 @@ def calculate_pval_from_trials(
 
     return (p, p_sigma)
 
+def polynomial_fit(
+        ns, p, p_weight, deg, p_thr):
+    """Performs a polynomial fit on the p-values of test-statistic trials
+    associated to each ns..
+    Using the fitted parameters and the covariance matrix it computes the
+    number of signal events correponding to the given p-value critical value
+    together with its uncertainty.
+    
+    Parameters
+    ----------
+    ns : 1D array_like object
+        x-coordinates of the sample.
+    p : 1D array_like object
+        y-coordinates of the sample.
+    p_weight : 1D array_like object
+        Weights to apply to the y-coordinates of the sample points. For gaussian
+        uncertainties, use 1/sigma.
+    deg : int
+        Degree of the fitting polynomial function.
+    p_thr : float within [0,1]
+        The critical p-value.
+
+    Returns
+    -------
+    ns : float
+    """
+    params, cov = np.polyfit(ns, p, deg, w=p_weight, cov=True)
+
+    params_var = []
+    for k in range(deg+1):
+        params_var.append(cov[k][k])
+
+    if deg == 1:
+        a,b = params[0],params[1]
+        a_var,b_var = params_var[0],params_var[1]
+        print('Fit params:',a,b)
+        print('Fit params variances:',a_var,b_var)
+
+        ns = (p_thr - b)/a
+
+        dnda = (b - p_thr)/(a**2)
+        dndb = -1/a
+
+        print('dnda:',dnda)
+        print('dndb:',dndb)
+        print('')
+
+        err = np.sqrt(dnda**2*a_var + dndb**2*b_var)
+        ns_err = np.sqrt((dnda**2*a_var)+(dndb**2*b_var))
+
+    if deg == 2:
+        a,b,c = params[0],params[1],params[2]
+        a_var,b_var,c_var = params_var[0],params_var[1],params_var[2]
+        print('Fit params:',a,b,c)
+        print('Fit params variances:',a_var,b_var,c_var)
+
+        ns = (- b + np.sqrt((b**2)-4*a*(c-p_thr))) / (2*a)
+
+        dnda = (0.5*(b*np.sqrt(b**2-4*(c-p_thr)*a)-b**2+2*a*(c-p_thr))/
+                    (a**2*np.sqrt(b**2-4*a*(c-p_thr))))
+        dndb = 1/(2*a)*(b/(np.sqrt(b**2-4*a*(c-p_thr)))-1)
+        dndc = -1/(np.sqrt(b**2-4*a*(c-p_thr)))
+
+        print('dnda:',dnda)
+        print('dndb:',dndb)
+        print('dndc:',dndc)
+        print('')
+
+        err = np.sqrt(dnda**2*a_var + dndb**2*b_var + dndc**2*c_var)
```

Comment:
You don't use the err anywhere, so it should be removed or put into a comment describing how it would be caluclated